core/libs/commonwealth/commonwealth/settings/manager.py

```diff
@@ -0,0 +1,120 @@
+import os
+import pathlib
+import re
+from typing import Any, Optional
+
+import appdirs
+from loguru import logger
+
+from commonwealth.settings import settings
+
+
+class Manager:
+    SETTINGS_NAME_PREFIX = "settings-"
+
+    def __init__(self, project_name: str, settings_type: Any, config_path: Optional[pathlib.Path] = None):
```

Comment:
I think `settings_type: typing.Type[settings.BaseSettings]` is the way to indicate for static checkers that `settings_type` inherits from `BaseSettings`.